O/ORTools/build_tarballs.jl

```diff
@@ -0,0 +1,181 @@
+    FileProduct("include/operations_research/math_opt/model.proto", :proto_math_opt_model),
+    FileProduct("include/operations_research/math_opt/model_parameters.proto", :proto_math_opt_model_parameters),
+    FileProduct("include/operations_research/math_opt/model_update.proto", :proto_math_opt_model_update),
+    FileProduct("include/operations_research/math_opt/result.proto", :proto_math_opt_result),
+    FileProduct("include/operations_research/math_opt/solution.proto", :proto_math_opt_solution),
+    FileProduct("include/operations_research/math_opt/parameters.proto", :proto_math_opt_parameters),
+    FileProduct("include/operations_research/math_opt/solvers/glpk.proto", :proto_math_opt_solvers_glpk),
+    FileProduct("include/operations_research/math_opt/solvers/gurobi.proto", :proto_math_opt_solvers_gurobi),
+    FileProduct("include/operations_research/math_opt/solvers/highs.proto", :proto_math_opt_solvers_highs),
+    FileProduct("include/operations_research/math_opt/sparse_containers.proto", :proto_math_opt_sparse_containers),
+    FileProduct("include/operations_research/math_opt/infeasible_subsystem.proto", :proto_math_opt_infeasible_subsystem),
+    # - From packing/
+    FileProduct("include/operations_research/packing/multiple_dimensions_bin_packing.proto", :proto_packing_multiple_dimensions_bin_packing),
+    FileProduct("include/operations_research/packing/vbp/vector_bin_packing.proto", :proto_packing_vector_bin_packing),
+    # - From pdlp/
+    FileProduct("include/operations_research/pdlp/solve_log.proto", :proto_pdlp_solve_log),
+    FileProduct("include/operations_research/pdlp/solvers.proto", :proto_pdlp_solvers),
+    # - From sat/
+    FileProduct("include/operations_research/sat/cp_model.proto", :proto_sat_cp_model),
+    FileProduct("include/operations_research/sat/sat_parameters.proto", :proto_sat_parameters),
+    FileProduct("include/operations_research/sat/boolean_problem.proto", :proto_sat_boolean_problem),
+    # FileProduct("include/operations_research/sat/v1/cp_model_service.proto", :proto_sat_v1_cp_model_service),  # RPC definition.
+    # - From util/
+    FileProduct("include/operations_research/optional_boolean.proto", :optional_boolean),
+    # - From scheduling/
+    FileProduct("include/operations_research/course_scheduling.proto", :proto_scheduling_course_scheduling),
+    FileProduct("include/operations_research/scheduling/rcpsp/rcpsp.proto", :proto_scheduling_rcpsp),
+    FileProduct("include/operations_research/scheduling/jssp/jobshop_scheduling.proto", :proto_scheduling_jobshop_scheduling),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"11", preferred_llvm_version=v"16", julia_compat="1.9")
```

Comment:
At the moment there's no LLVM 16 in BinaryBuilder, and contrary to GCC for LLVM we use the _latest_ version available anyway
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"11", julia_compat="1.9")
```